gulp/tasks/test.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+var gulp = require( 'gulp' );
+var $ = require( 'gulp-load-plugins' )();
+var protractor = require( 'gulp-angular-protractor' );
+var exec = require( 'child_process' ).exec;
+var config = require( '../config' ).test;
+
+gulp.task( 'test:unit', function( cb ) {
+  gulp.src( config.src )
+    .pipe( $.istanbul( {
+      includeUntested: true
+    } ) )
+    .pipe( $.istanbul.hookRequire() )
+    .on( 'finish', function() {
+      gulp.src( config.tests + '/unit_tests/**/*.js' )
+        .pipe( $.mocha( {
+          reporter: 'nyan'
+        } ) )
+        .pipe( $.istanbul.writeReports( {
+          dir: config.tests + '/unit_test_coverage'
+        } ) )
+
+        /* we want this but it breaks because we don't have good coverage
```

Comment:
I was sad commenting it out, but I want to keep it in here so we don't forget about it